### ISSUE TITLE
Excessive render of toggled icon

### ISSUE THREAD

**Issue by jakublacko** [alpinejs/alpine#2423]
### Discussed in https://github.com/alpinejs/alpine/discussions/2351

<div type='discussions-op-text'>

<sup>Originally posted by **MarioPerini** November 10, 2021</sup>
I use Fontawesome Icons as SVG's and have a performance issue when using it combined with alpine.

Here is a simple example:
https://codepen.io/MarioPerini/pen/ZEJjYzj

If you look at chrome-dev console you will see a constant re-render of the parent and the icon svg.
![image](https://user-images.githubusercontent.com/40631006/141121756-8369dd01-d63d-44d7-8fc2-84abac74edfd.png)

This causes a significant load on CPU and GPU. I opened the tab about half-time of the chart:
![image](https://user-images.githubusercontent.com/40631006/141122423-8a4056c5-9e58-4dff-a823-44fc7e8d6454.png)

Expected behaviour:
Render once and re-render only on click
</div>

**Comment by danddanddand** [comment]
I investigated a bit and I think you just can't dynamically change the icon class within the <i></i> because FontAwesome is setting up a MutationObserver on that tag and replacing it with the SVG and basically creating infinite loop with Alpine's MutationObserver because they are acting on the same element.

In this codepen https://codepen.io/danddanddand/pen/rNGNjJL I have two versions that are safe because MutationObservers aren't fighting over same element and a commented out unsafe example using x-effect that will cause same issue your above example does but for demonstration purposes might help you understand what's happening.

**Comment by danddanddand** [comment]
@jakublacko Can we close this?

**Comment by jakublacko** [comment]
I would like to say no, since it is not fixed, this is just a workaround.
Can we think about some possiblity of disabling font awesome observer or alpine observer for this case?
There must be some way how to do it really properly

**Comment by jakublacko** [comment]
For example for Vue.js it is solved by own observer, maybe it would be possible to take a look on their solution and implement something similiar for Alpine.js?

**Comment by SimoTod** [comment]
Have you tried asking fontawesome if they have a solution? Looking at their website, they provide custom plugins to use it with third party libs like Alpine.

I don't see it being fixed in Alpine because it would require an adhoc implementation for fontawesome inside the Alpine core which sounds odd. It would also add kb to the library for something reported only by one user so far. 

Maybe you can write a custom directive (x-fa) with a specific implementation to avoid that bug.

**Comment by danddanddand** [comment]
It’s also only an issue with the specific way you’re using font awesome. I’ve used font awesome with Alpine for long time using CSS version but the SVG JS version would work fine in my project because icons are selected server side and `<i></i>` tags with font awesome family and font are never toggle directly or styled directly, they are always wrapped in a span and toggled by toggling entire tag not x-binding the class.

**Comment by SimoTod** [comment]
Yeah, i believe the issue is that the generated svg copies all the attributes from the original `i` tag so alpine mutation observer reevaluates :class, triggering a mutation that is picked up by the fa mutation observer that updates the svg tag triggering another mutation picked up by alpine in loop.
